### PR TITLE
Misc

### PR DESCRIPTION
**Pull request by bbenligiray**
Addresses API3-12, selectively

What I didn't address:
- The typecasting is indeed surprisingly expensive (~50 gas based on my tests). However, in addition to struct-packing reasons, using `uint32` in interfaces is useful in that it enforces additional type checks. There are only a few points where this optimization makes sense (because a lot of the time the timestamp will needed to be packed in a struct), and imo they are not worth creating special cases.
- We avoid custom errors because we may deploy on weird chains with custom block explorers that don't support them. We're not overly concerned about deployment costs (because our contracts are deployed once per chain) and optimizing the gas cost of the reverting branch is not a significant concern.
- We don't see a reason to limit role description lengths to 32 characters, and some of our role description lengths are already longer than that. Also, the fact that they are of string type is only a gas cost issue because we decided to provide the full role descriptions on-chain (because we will deploy these only once per chain so the additional gas cost is worth it). An AccessControlRegistry user implementation that is concerned about the additional gas cost should hardcode the description hashes instead.
- Dividing `require()` conditions that will result in the same error message being emitted reduces readability and the gas cost saving isn't worth it.

### CODE REVIEW

File: contracts/dapis/DapiServer.sol [hunk 1]
```diff
@@ -572,23 +575,15 @@ contract DapiServer is
...
     /// matter for the purposes of a Beacon set update subscription.
-    /// @param subscriptionId Subscription ID
-    /// @param airnode Airnode address
-    /// @param relayer Relayer address
-    /// @param sponsor Sponsor address
-    /// @param timestamp Timestamp used in the signature
     /// @param data Fulfillment data (an `int256` encoded in contract ABI)
-    /// @param signature Subscription ID, timestamp, sponsor wallet address
-    /// (and fulfillment data if the relayer is not the Airnode) signed by the
-    /// Airnode wallet
```
Comment (acenolaza) [inline contracts/dapis/DapiServer.sol:583]: WDYT of keeping the references like this:
```
    /// @param // subscriptionId Subscription ID
    /// @param // airnode Airnode address
    /// @param // relayer Relayer address
    /// @param // sponsor Sponsor address
    /// @param // timestamp Timestamp used in the signature
    /// @param data Fulfillment data (an `int256` encoded in contract ABI)
    /// @param // signature Subscription ID, timestamp, sponsor wallet address
    /// (and fulfillment data if the relayer is not the Airnode) signed by the
    /// Airnode wallet
```

File: contracts/dapis/DapiServer.sol [hunk 2]
```diff
@@ -572,23 +575,15 @@ contract DapiServer is
...
-        address relayer, // solhint-disable-line no-unused-vars
-        address sponsor, // solhint-disable-line no-unused-vars
-        uint256 timestamp, // solhint-disable-line no-unused-vars
+        bytes32,
+        address,
+        address,
+        address,
+        uint256,
         bytes calldata data,
-        bytes calldata signature // solhint-disable-line no-unused-vars
```
Comment (acenolaza) [inline contracts/dapis/DapiServer.sol:591]: And we can even keep the argument names like this for reference:

```
    function fulfillPspBeaconSetUpdate(
        bytes32 /*subscriptionId*/,
        address /*airnode*/,
        address /*relayer*/,
        address /*sponsor*/,
        uint256 /*timestamp*/,
        bytes calldata data,
        bytes calldata /*signature*/
```

**Review by acenolaza — APPROVED** [review]
LGTM 👍🏻

**Comment by Ashar2shahid** [comment]
> The typecasting is indeed surprisingly expensive (~50 gas based on my tests). However, in addition to struct-packing reasons, using uint32 in interfaces is useful in that it enforces additional type checks. There are only a few points where this optimization makes sense (because a lot of the time the timestamp will needed to be packed in a struct), and imo they are not worth creating special cases.

Yeah I was looking into it a while ago, apparently casting to int256/uint256 is the cheapest (2 gas) as seen [here ](https://ethereum.stackexchange.com/questions/119947/what-are-the-gas-costs-for-type-casting-or-conversions)

> Dividing require() conditions that will result in the same error message being emitted reduces readability and the gas cost saving isn't worth it.

IDK, it makes the revert messages more specific(we don't have to reuse the same message in both the reverts) and saves gas too. might be worth it

**Review by Ashar2shahid — APPROVED** [review]

File: contracts/dapis/DapiServer.sol (hunk 1 above)
Comment (bbenligiray) [reply contracts/dapis/DapiServer.sol]: Strongly agree, did this

File: contracts/dapis/DapiServer.sol (hunk 2 above)
Comment (bbenligiray) [reply contracts/dapis/DapiServer.sol]: and this

**Comment by bbenligiray** [comment]
If the conditions warranted different revert strings they would have been divided across requires in the first place

**Pull request merged**